tflearn/layers/recurrent.py

```diff
@@ -514,7 +515,13 @@ def __call__(self, inputs, state, scope=None):
             new_c = (c * self._inner_activation(f + self._forget_bias) +
                      self._inner_activation(i) *
                      self._activation(j))
-            new_h = self._activation(new_c) * self._inner_activation(o)
+            
+            # hidden-to-hidden batch normalizaiton
+            if batch_norm == False:
+                new_h = self._activation(new_c) * self._inner_activation(o)
+            else:
+                batch_norm_new_c = batch_normalization(new_c, restore=False)
```

Comment:
here you can propagate arguments to batch_normalization: `trainable=self.trainable`, `restore=self.restore`, `reuse=self.reuse`, so that it can works synchronously with the Weights same parameters.